assets/js/plan_template.js

```diff
@@ -0,0 +1,175 @@
+    }
+}
+
+function insertNewRow(start = '8', end = '10', category = 'Visit') {
+    const template = document.getElementById("template");
+    let newRow = template.insertRow(-1);
+    let newCell = newRow.insertCell(0);
+    let newCell2 = newRow.insertCell(1);
+    let newCell3 = newRow.insertCell(2);
+
+    let select = document.createElement("select");
+    let cafeOption = document.createElement("option");
+    let attractionOption = document.createElement("option");
+
+    let startTime = hourDropdown();
+    startTime.classList.add("form-select");
+    startTime.value = start;
+
+    let endTime = hourDropdown();
+    endTime.classList.add("form-select");
+    endTime.value = end;
+
+    cafeOption.text = "Eatery";
+    attractionOption.text = "Visit";
+
+    select.classList.add("form-select");
+    select.add(cafeOption);
+    select.add(attractionOption);
+    select.value = category;
+
+    newCell.append(select);
+
+    newCell2.append(startTime);
+
+    newCell3.append(endTime);
+}
+
+function hourDropdown() {
+    let select = document.createElement("select");
+    for (let hour = 8; hour < 20; hour++) {
+        let option = document.createElement("option");
+        option.text = hour.toString();
+        select.add(option);
+    }
+    return select;
+}
+
+function tableToJSON() {
+    return $("#template").tableToJSON(
+        {
+            extractor: function ($cellIdx, $cell) {
+                return $cell.find("select").val()
+            }
+        }
+    )
+}
+
+function tableToSlots() {
+    const rows = tableToJSON()
+    return rows.map(
+        function (row) {
+            return {
+                "category": row["Category"],
+                "time_slot": {
+                    "slot": {
+                        "start": parseInt(row["Start"]),
+                        "end": parseInt(row["End"])
+                    }
+                }
+            }
+        }
+    )
+}
+
+async function postTemplate() {
+    const location = document.getElementById('location').value.toString();
+    const locationFields = location.split(",");
+    let locationToPost = {}
+    switch (locationFields.length) {
+        case 2:
+            locationToPost = {
+                "city": locationFields[0],
+                "country": locationFields[1]
+            }
+        case 3:
+            locationToPost = {
+                "city": locationFields[0],
+                "adminAreaLevelOne": locationFields[1],
+                "country": locationFields[2]
+            }
+    }
+
+    const data = {
+        "location": locationToPost,
+        "slots": tableToSlots()
+    }
+    const url = "/v1/customize";
+    console.log(`data about to send: ${JSON.stringify(data)}`);
+    await fetch(
+        url, {
+        method: "POST",
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+    }
+    ).then(response => response.json())
+        .then(
```

Comment:
Are we using this promise chain with a series of `then()` in JS? https://javascript.info/promise-chaining
If so, how about use a different variable name such as `data` instead of `response` in line 123?

Line 123 could be `data => parseResponse(data)`.

Reply:
We should always use `axios`! 👍🏼